app/forms/nsm/youth_court_fee_form.rb

```diff
@@ -0,0 +1,48 @@
+module Nsm
+  class YouthCourtFeeForm < BaseAdjustmentForm
+    COMMENT_FIELD = 'youth_court_fee_adjustment_comment'.freeze
+    LINKED_CLASS = V1::AdditionalFee
+
+    attribute :remove_youth_court_fee
+    attribute :youth_court_fee_adjustment_comment
+
+    def save
+      return false unless valid?
+
+      remove_bool = ActiveModel::Type::Boolean.new.cast(remove_youth_court_fee)
+
+      process_field(value: !remove_bool, field: 'include_youth_court_fee')
+
+      true
+    end
+
+    private
+
+    def ensure_original_field_value_set(field)
+      if selected_record["#{field}_original"].present?
+        selected_record.delete("#{field}_original")
+        selected_record.delete(self.class::COMMENT_FIELD)
+      else
+        selected_record["#{field}_original"] ||= selected_record[field]
+      end
+    end
+
+    def explanation_required?
+      remove_youth_court_fee == 'true'
+    end
+
+    def selected_record
+      @selected_record ||= submission.data
+    end
+
+    def data_has_changed?
+      remove_youth_court_fee != item.include_youth_court_fee
```

Comment:
This looks suspect, for 2 reasons:

First, I believe `remove_youth_court_fee` is a string, so comparing it with a boolean will go all kinda wokny.

Second, if it was a boolean,  if there was a youth court fee, and I wanted to remove it, then we'd be evaluating `true != true` which will return false, so this would suggest that the data has _not_ changed, even though I was trying to change the data.

Reply:
do we even need to override this method since explanation is overidden?